sys/kern/kern_cheri_revoke.c

```diff
@@ -574,6 +579,17 @@ kern_cheri_revoke_get_shadow(struct thread *td, int flags,
 	if (!cheri_gettag(cres))
 		return (EINVAL);
 
+	vm = vmspace_acquire_ref(td->td_proc);
```

Comment:
Is `td` always curthread here? If so I think there's no need to grab a reference.

Reply:
Should always be curthread. I'll add an assert. The rest of the function somewhat inappropriately assumes curproc is ok so the assert seems like a generally good idea.